astropy/wcs/utils.py

```diff
@@ -1023,11 +1023,14 @@ def fit_wcs_from_points(xy, world_coords, proj_point='center',
         lon, lat = unit_sph.lon.deg, unit_sph.lat.deg
 
     # verify input
-    if (proj_point != 'center') and (type(proj_point) != type(world_coords)):
+    if (type(proj_point) != type(world_coords)) and (proj_point != 'center'):
```

Comment:
What is the reason for using `type` instead of `isinstance`?

Regardless of the answer, however, I am not sure this is the right place to fix this. Shouldn't the fix actually be done in `SkyCoord.__eq__()`? CC: @eteq

Reply:
> What is the reason for using `type` instead of `isinstance`?

Not sure, that is what was used before and I wanted to keep the changes for the fix minimal. I thought refactoring might be better done in a separate issue/PR.

Reply:
I don't understand why the change in this line, I wouldn't have thought it mattered what order the checks are in since it's an `and` statement. What am I missing?

Reply:
> I don't understand why the change in this line, I wouldn't have thought it mattered what order the checks are in since it's an `and` statement. What am I missing?

Because of lazy evaluation: If `proj_point` has the same type as the `world_coords` - so probably a `SkyCoord` - the second check `proj_point != 'center'` isn't executed and therefore won't throw the error anymore.

Reply:
> > What is the reason for using `type` instead of `isinstance`?
> 
> Not sure, that is what was used before and I wanted to keep the changes for the fix minimal. I thought refactoring might be better done in a separate issue/PR.

I would prefer to improve things, if we can... I was thinking of replacing this line and the next couple of lines with

```
    if isinstance(proj_point, str) and proj_point == 'center':
        use_center_as_proj_point = True
    elif isinstance(proj_point, SkyCoord):
        use_center_as_proj_point = False
        if proj_point.size != 1:
            raise ValueError("'proj_point' must contain exactly one point")
    else:
            raise ValueError("proj_point must be set to 'center', or an"
                             "`~astropy.coordinates.SkyCoord` object with "
                             "a pair of points.")
```

But looking at the rest of the code, I see that there are other instances that need refactoring and so I would agree with you.